src/Authentication/Authentication/Common/GraphProfile.cs

```diff
@@ -0,0 +1,61 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph.PowerShell.Authentication.Common
+{
+    using System;
+    using System.Management.Automation;
+    using System.Linq;
+    using System.Collections;
+    using System.Collections.ObjectModel;
+    using System.IO;
+
+    /// <summary>
+    /// Methods for working with Microsoft Graph profiles.
+    /// </summary>
+    internal static class GraphProfile
+    {
+        public static string[] GetProfiles(CommandInvocationIntrinsics invokeCommand, bool listAvailable = false, params string[] moduleNames)
+        {
+            return GetModules(invokeCommand, listAvailable, moduleNames).SelectMany(GetProfiles).Distinct().ToArray();
+        }
+
+        public static PSModuleInfo[] GetModules(CommandInvocationIntrinsics invokeCommand, bool listAvailable = false, params string[] moduleNames)
+        {
+            string nameParameter = $" -Name { (moduleNames != null && moduleNames.Any() ? GetCommaSeparatedQuotedList(moduleNames) : "Microsoft.Graph*" )}";
```

Comment:
NIT: might cause less future mistakes to put the space in the command string declaration rather than here